x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals_histogram_panel/signals_histogram/index.tsx

```diff
@@ -46,18 +50,42 @@ export const SignalsHistogram = React.memo<HistogramSignalsProps>(
     filters,
     legendPosition = 'right',
     loadingInitial,
+    registerQuery,
+    signalIndexName,
     setTotalSignalsCount,
     stackByField,
     updateDateRange,
   }) => {
-    const [isLoadingSignals, signalsData, setQuery] = useQuerySignals<{}, SignalsAggregation>(
-      getSignalsHistogramQuery(stackByField, from, to, [])
+    const {
+      loading: isLoadingSignals,
+      data: signalsData,
+      setQuery,
+      response,
+      request,
+      refetch,
+    } = useQuerySignals<{}, SignalsAggregation>(
+      getSignalsHistogramQuery(stackByField, from, to, []),
+      signalIndexName
     );
     const theme = useTheme();
     const kibana = useKibana();
 
     const formattedSignalsData = useMemo(() => formatSignalsData(signalsData), [signalsData]);
 
+    useEffect(() => {
+      if (refetch != null && registerQuery != null) {
+        registerQuery({
+          id: DETECTIONS_HISTOGRAM_ID,
+          inspect: {
+            dsl: [request],
+            response: [response],
+          },
+          loading: isLoadingSignals,
+          refetch: refetch as inputsModel.Refetch,
```

Comment:
I was able to remove the `as` without any linter error

```suggestion
          refetch,
```